fabric/Makefile

```diff
@@ -0,0 +1,92 @@
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+TOP = ..
+include $(TOP)/build.mk
+
+FABRIC_VERSION ?= v2.0.1
+FABRIC_PEER_BIN = $(FABRIC_PATH)/build/bin/peer
+FABRIC_TLCC = $(FABRIC_PATH)/core/scc/tlcc
+
+FABRIC_CURRENT_BRANCH = $(shell cd $(FABRIC_PATH); git rev-parse --abbrev-ref HEAD)
+FABRIC_CURRENT_TAG = $(shell cd $(FABRIC_PATH); git describe --tags)
+FABRIC_PATCHED_BRANCH = $(FABRIC_VERSION)-fpc
+FABRIC_PATCHED = $(FABRIC_PATH)/.fpc_patched
+FABRIC_TRACKED_CHANGES = $(shell cd $(FABRIC_PATH); git status -s | grep -v '\?')
+FABRIC_COMMITS = $(shell cd $(FABRIC_PATH); git log $(FABRIC_VERSION)..HEAD --oneline)
+FABRIC_LOG_DIFF = $(shell diff $(FABRIC_PATCHED) <(FABRIC_COMMITS))
+
+build: native
+
+patch: $(FABRIC_PATCHED)
+
+$(FABRIC_PATCHED):
+	@echo "Patching Fabric ..."
+	@if [ "$(FABRIC_CURRENT_BRANCH)" != "$(FABRIC_PATCHED_BRANCH)" ]; then \
+		echo "Aborting! Current branch is $(FABRIC_CURRENT_BRANCH) but want $(FABRIC_PATCHED_BRANCH)"; exit 1; \
+	fi
+
+	@if [ "$(FABRIC_CURRENT_TAG)" != "$(FABRIC_VERSION)" ]; then \
+		echo "Aborting! Current branch HEAD does not match tag/$(FABRIC_VERSION)! $(FABRIC_CURRENT_TAG) but want $(FABRIC_VERSION)"; exit 1; \
+	fi
+
+	cd $(FABRIC_PATH) && \
+	git am $(FPC_PATH)/fabric/*.patch && \
+	git log $(FABRIC_VERSION)..HEAD --oneline > $@
+
+clean-patch:
+	@echo "Cleaning Fabric patches ..."
+	@if [ "$(FABRIC_CURRENT_BRANCH)" != "$(FABRIC_PATCHED_BRANCH)" ]; then \
+		echo "Aborting! Current branch is $(FABRIC_CURRENT_BRANCH) but want $(FABRIC_PATCHED_BRANCH)"; \
+		echo "Please checkout $(FABRIC_PATCHED_BRANCH) and come back."; \
+		exit 1; \
+	fi
+
+	@if [ "$(FABRIC_CURRENT_TAG)" = "$(FABRIC_VERSION)" ]; then \
+		echo "Aborting! Nothing to do here ... it seems you haven't applied the patches yet"; exit 1; \
+	fi
```

Comment:
hmm, exit 1 for this condition is ugly, you shouldn't terminate make if you run it in an already cleaned setup? i think it should be the inversion of the test and then everything subsequent inside?

Reply:
I agree! This would be better - I've learnt that a exit 0 does not work in this case an there is not wait to abort without an error. As agreed we can improve this in another PR.